src/AWS.Deploy.Common/Recommendation.cs

```diff
@@ -118,11 +117,7 @@ public OptionSettingItem GetOptionSetting(string? jsonPath)
                 }
                 if (optionSetting.Type.Equals(OptionSettingValueType.KeyValue))
                 {
-                    if (i + 2 == ids.Length)
-                        return optionSetting;
-                    else
-                        throw new OptionSettingItemDoesNotExistException(DeployToolErrorCode.OptionSettingItemDoesNotExistInRecipe, $"The Option Setting Item {jsonPath} does not exist as part of the" +
-                    $" {Recipe.Name} recipe");
+                    return optionSetting;

```

Comment:
Both changes appear to be unit testable. Is there something I'm overlooking?
It would be good to know that this function is working as intended, and doesn't introduce a regression when updated in the future.

Reply:
I added some unit tests